packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -37,7 +41,12 @@ export function startResourceCollection(
   sessionManager: RumSessionManager
 ) {
   lifeCycle.subscribe(LifeCycleEventType.REQUEST_COMPLETED, (request: RequestCompleteEvent) => {
-    lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processRequest(request, configuration, sessionManager))
+    waitForResponseToFinish(request, () =>
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why not waiting for response to finish inside the `requestCollection`?

Reply:
Good point, I initially put it here as this was where `PerformanceEntry` logic resided. But bar that there was no real reason for it to be here.

It probably is better for us to move this logic into `requestCollection` as it is closer to the instrumentation and thus will give us a better `duration` estimation

Reply:
If we put it in `requestCollection` and wait For Response To Finish  before triggering `REQUEST_COMPLETED` it will also affect `waitPageActivityEnd`. Conceptually it would make sens to me but might have broader impact. @BenoitZugmeyer wdyt?

Reply:
Makes sense to me too! The UI usually react when the response finishes to load